node_lambnik/src/tiler/src/tiler.js

```diff
@@ -129,29 +94,27 @@ const encodeAsPNG = (renderedTile) => new Promise((resolve, reject) => {
  * @param z
  * @param x
  * @param y
- * @returns {mapnik.Map}
+ * @returns {Promise<mapnik.Map>}
  */
 const createMap = (z, x, y) => {
-    // Get latlng bounds for zxy tile
-    const bounds = tileBounds(z, x, y)
-
-    // Create a webmercator map
-    const map = new mapnik.Map(TILE_WIDTH, TILE_HEIGHT, PROJECTION)
+    // Create a webmercator map with specified bounds
+    const map = new mapnik.Map(TILE_WIDTH, TILE_HEIGHT)
     map.bufferSize = 64
-    map.extent = bounds
-    const layer = new mapnik.Layer('tile', PROJECTION)
-    layer.styles = ['point']
+    map.extent = new SphericalMercator().bbox(x, y, z, false, process.env.EPSG)
```

Comment:
This is still trying to load the projection from the environment.  It should go below, inside the `map.fromString` callback, and use the projection loaded from the config, i.e. modify the map with
```
result.extent = new SphericalMercator().bbox(x, y, z, false, result.srs)
```
before calling `resolve`.  (I might also vote for a more descriptive variable name than `result`, e.g. `configuredMap`.)

Reply:
Upon further inspection, that last argument isn't doing anything anyway.  The argument is called `srs` [in the docs](https://github.com/mapbox/sphericalmercator#bboxx-y-zoom-tms_style-srs), but it doesn't actually support arbitrary projections.  It just [checks for the value `'900913'`](https://github.com/mapbox/sphericalmercator/blob/master/sphericalmercator.js#L116) (the [unofficial/non-EPSG id of the web mercator projection](https://en.wikipedia.org/wiki/Web_Mercator#Identifiers)) and converts to that if it matches.  Otherwise it returns WGS84 lat/lon (aka EPSG 4326).

So yeah, the [CartoCSS config uses PROJ4 format](https://cartocss.readthedocs.io/en/latest/mml.html#srs) so if we do need to get `SphericalMercator` to convert the bbox to Web Mercator we'd need to parse the configured SRS and pass the 900913 ID instead.  If we need to support other projections, I think we'd need to add logic to detect the incoming SRS and reproject.

I think we're OK, though--that we can just remove the last argument from this function call and only support lat/lon geometries for now.  Which means we could move it inside the callback, but for now it will work the same in either place.

Reply:
Interesting that the `srs` parameter actually handle projections in the way I assumed it did. I'll move that line into the callback for future reference.